File: test-suite/src/test/java/io/micronaut/email/docs/EmailSenderTest.java
package io.micronaut.email.docs;

import io.micronaut.context.BeanContext;
import io.micronaut.email.EmailSender;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(startApplication = false)
public class EmailSenderTest {

    @Inject
    BeanContext beanContext;

    @Test
    void beanOfTypeEmailSenderExists() {
        assertTrue(beanContext.containsBean(EmailSender.class));
    }
}

File: test-suite-python/src/test/python/micronaut/email/docs/EmailSenderTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import BeanContext
from micronaut.email import EmailSender
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest(startApplication=False)
class EmailSenderTest:
    bean_context: Annotated[BeanContext, Inject]

    @Test
    def bean_of_type_email_sender_exists(self):
        assert self.bean_context.containsBean(EmailSender)
